src/workerd/api/kv-instrumentation-test.js

```diff
@@ -22,26 +22,37 @@ export default {
 
     // Accumulate the span info for easier testing
     return (event) => {
-      // span ids are simple counters for tests, but invocation ID allows us to differentiate them
```

Comment:
I think the changes here duplicate code, reduce legibility and cause us to no longer throw if we encounter spans that fail to be closed or attribute/spanClose events with a rogue spanId – maybe we can just filter out and ignore top-level attributes events? That should be a lot cleaner, just need to identify them in the attributes case based on their spanId matching the top-level spanId.